lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp

```diff
@@ -754,4 +769,55 @@ TEST_F(AMFAppProcedureTest, TestPDUSession_Invalid_PDUSession_Identity) {
   EXPECT_TRUE(expected_Ids == AMFClientServicer::getInstance().msgtype_stack);
 }
 
+TEST_F(AMFAppProcedureTest, TestRegistrationProcSUCIExt) {
+  amf_ue_ngap_id_t ue_id = 0;
+
+  // Send the initial UE message
+  imsi64_t imsi64 = 0;
+  int rc          = RETURNok;
+  imsi64          = send_initial_ue_message_no_tmsi(
+      amf_app_desc_p, 36, 1, 1, 0, plmn, intital_ue_message_suci_ext_hexbuf,
+      sizeof(intital_ue_message_suci_ext_hexbuf));
+
+  rc = amf_decrypt_imsi_info_answer(&decrypted_imsi);
+  EXPECT_TRUE(rc == RETURNok);
+
+  // Check if UE Context is created with correct imsi
+  bool res = false;
+  res      = get_ue_id_from_imsi(amf_app_desc_p, imsi64, &ue_id);
+  EXPECT_TRUE(res == true);
+
+  // Send the authentication response message from subscriberdb
+  rc = send_proc_authentication_info_answer(imsi, ue_id, true);
+  EXPECT_TRUE(rc == RETURNok);
+
+  // Validate if authentication procedure is initialized as expected
+  res = validate_auth_procedure(ue_id, 0);
+  EXPECT_TRUE(res == true);
+
+  // Send uplink nas message for auth response from UE
+  rc = send_uplink_nas_message_ue_auth_response(
+      amf_app_desc_p, ue_id, plmn, ue_auth_response_hexbuf,
+      sizeof(ue_auth_response_hexbuf));
+  EXPECT_TRUE(rc == RETURNok);
+
+  // Check whether security mode procedure is initiated
+  res = validate_smc_procedure(ue_id, 0);
+  EXPECT_TRUE(res == true);
+
+  // Send uplinkg nas message for security mode complete response from UE
+  rc = send_uplink_nas_message_ue_smc_response(
+      amf_app_desc_p, ue_id, plmn, ue_smc_response_hexbuf,
+      sizeof(ue_smc_response_hexbuf));
+  EXPECT_TRUE(rc == RETURNok);
+
+  // Send uplink nas message for registration complete response from UE
+  rc = send_uplink_nas_registration_complete(
+      amf_app_desc_p, ue_id, plmn, ue_registration_complete_hexbuf,
+      sizeof(ue_registration_complete_hexbuf));
+  EXPECT_TRUE(rc == RETURNok);
+
+  amf_app_handle_deregistration_req(ue_id);
+}
+
 }  // namespace magma5g
```

Comment:
@ulaskozat As suggested, added the UT code for registration procedure for the impact of SUCI Extensions. Please review and share your comments to proceed onto for approval. Thanks.
CC @panyogesh